### PR TITLE
Rename Labels to Requests

### PR DESCRIPTION
**Pull request by sprjr**
@jmeas this is a bit of an early PR, and probably needs some squashing and git clean up - but I think is a good start on renaming / removing some of the labels stuff from the repo.

Mind taking an early look? The tests are *all* broken, and I haven't started on documentation - just wanting to keep on the right track for this stuff. Might be useful to make a list in this PR for something things todo so we can close out #161.

### CODE REVIEW

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js [hunk 1]
```diff
@@ -76,36 +71,15 @@ function delSucceed(state, action, {initialResourceMeta}) {
...
-    };
-
-    if (hasIds && existingLabel.ids) {
-      newLabel.ids = existingLabelIds.filter(r => !idList.includes(r));
-    } else if (existingLabel.ids) {
-      newLabel.ids = existingLabelIds;
-    }
-
-    if (label && label === requestLabel) {
-      newLabel.status = requestStatuses.SUCCEEDED;
```
Comment (jmeas) [inline packages/resourceful-redux/src/request-statuses-plugin/delete.js]: We need this still. Labels still track successful request responses.

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js [hunk 2]
```diff
@@ -76,36 +71,15 @@ function delSucceed(state, action, {initialResourceMeta}) {
 
   const hasIds = idList && idList.length;
 
-  // If we have no, IDs, list, nor request label, then there is nothing to update
-  if (!hasIds && !label) {
```
Comment (jmeas) [inline packages/resourceful-redux/src/request-statuses-plugin/delete.js]: Still need this

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js [hunk 3]
```diff
@@ -10,11 +10,6 @@ const delNull = reducerGenerator('delete', requestStatuses.NULL);
 function delSucceed(state, action, {initialResourceMeta}) {
   const resources = action.resources;
 
-  let label;
-  if (action.label && typeof action.label === 'string') {
-    label = action.label;
-  }
```
Comment (jmeas) [inline packages/resourceful-redux/src/request-statuses-plugin/delete.js]: Still need this

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js [hunk 4]
```diff
@@ -150,7 +124,6 @@ function delSucceed(state, action, {initialResourceMeta}) {
     ...state,
     meta: newMeta,
     lists: newLists,
-    labels: newLabels,
```
Comment (jmeas) [inline packages/resourceful-redux/src/request-statuses-plugin/delete.js]: Still need this

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js (hunk 3 above)
Comment (sprjr) [reply packages/resourceful-redux/src/request-statuses-plugin/delete.js]: crap sry

File: packages/resourceful-redux/src/utils/get-resources.js [hunk 5]
```diff
@@ -1,5 +1,4 @@
-// Returns a list of resources by IDs or label
-export default function(state, resourceName, idsOrLabel) {
+export default function(state, resourceName, idsOrRequest) {
```
Comment (jmeas) [inline packages/resourceful-redux/src/utils/get-resources.js]: `idsOrList` :)

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js [hunk 6]
```diff
@@ -76,35 +76,28 @@ function delSucceed(state, action, {initialResourceMeta}) {
 
   const hasIds = idList && idList.length;
 
-  // If we have no, IDs, list, nor request label, then there is nothing to update
-  if (!hasIds && !label) {
+  // If we have no IDs, then there is nothing to update
```
Comment (jmeas) [inline packages/resourceful-redux/src/request-statuses-plugin/delete.js]: I think we accidentally deleted part of this comment. Can we go back to the old comment but with `request label` swapped with `request name`?

File: packages/resourceful-redux/src/resource-reducer.js [hunk 7]
```diff
@@ -45,10 +45,10 @@ export default function resourceReducer(resourceName, options = {}) {
 
   return function reducer(state = initial, action) {
     if (process.env.NODE_ENV !== 'production') {
-      if (action.label && typeof action.label !== 'string') {
+      if (action.request && typeof action.request !== 'string') {
         warning(
-          `An invalid label was included in an action with type ` +
-          `"${action.type}". Labels must be strings.`
+          `An invalid request was included in an action with type ` +
```
Comment (jmeas) [inline packages/resourceful-redux/src/resource-reducer.js]: "request name"

File: packages/resourceful-redux/src/resource-reducer.js [hunk 8]
```diff
@@ -45,10 +45,10 @@ export default function resourceReducer(resourceName, options = {}) {
...
+          `"${action.type}". Requests must be strings.`
```
Comment (jmeas) [inline packages/resourceful-redux/src/resource-reducer.js]: "Request names"

File: packages/resourceful-redux/src/utils/cru-reducer-helper.js [hunk 9]
```diff
@@ -40,8 +40,8 @@ export default function(state, action, {initialResourceMeta}, updatedMeta) {
     }
   }
 
-  // Without resources or labels, there is nothing to update
-  if (!hasResources && !label && !list) {
+  // Without resources, lists, or requests, there is nothing to update
```
Comment (jmeas) [inline packages/resourceful-redux/src/utils/cru-reducer-helper.js]: `// Without resources, a list, or a request name, there is nothing to update`

list/name are singular for actions

File: packages/resourceful-redux/src/utils/reducer-generator.js [hunk 10]
```diff
@@ -8,19 +8,19 @@ import warning from './warning';
 // Basically, two things can change:
 //
 // 1. Request status for resource IDs in `meta`, if IDs are passed in
-// 2. Request status for a labeled request
+// 2. Request status for a named request
 //
-// A label's IDs don't change, and neither does the resource. Consequently, this
-// helper completely defines all of the ways in which the non-success reducers
+// A named requests's IDs don't change, and neither does the resource. Consequently,
```
Comment (jmeas) [inline packages/resourceful-redux/src/utils/reducer-generator.js]: `A named request's IDs don't change`

File: packages/resourceful-redux/src/utils/get-resources.js (hunk 5 above)
Comment (sprjr) [reply packages/resourceful-redux/src/utils/get-resources.js]: EFF. Thx.

File: packages/resourceful-redux/src/request-statuses-plugin/delete.js (hunk 6 above)
Comment (sprjr) [reply packages/resourceful-redux/src/request-statuses-plugin/delete.js]: Yea, it originally was not correct (included `list` but not in the logic) and then when I took out the `label` check I didn't add it back in for `request`. Whoops.

File: packages/resourceful-redux/src/utils/reducer-generator.js (hunk 10 above)
Comment (sprjr) [reply packages/resourceful-redux/src/utils/reducer-generator.js]: might be missing this but the comment says exactly what your comment says?

**Comment by sprjr** [comment]
Ok, should have all these comments covered and squashed back into their original commits. We can re-git those however you'd like. I'm going to update the tests next then I _think_ this is g2g.

File: packages/resourceful-redux/src/utils/reducer-generator.js (hunk 10 above)
Comment (jmeas) [reply packages/resourceful-redux/src/utils/reducer-generator.js]: requests’s => request’s

**Comment by sprjr** [comment]
@jmeas so I've rebased this branch down to squash some of the commits - let me know if you want any of the remaining ones to be squashed further.

There are two addition `fixup!` commits that update the delete tests to include that negative test, and the read tests to remove some duplicate tests that only applies to lists now.

I don't _see_ `label` in the repo anywhere, so let me know what else you'd like to see updated for this PR and I'll update #196 with a todo list and try to wrap that up.

Thanks!
